src/gt4py/next/program_processors/runners/dace.py

```diff
@@ -0,0 +1,128 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import functools
+from typing import cast
+
+import dace
+import factory
+from dace.codegen.compiled_sdfg import CompiledSDFG
+
+import gt4py._core.definitions as core_defs
+from gt4py.next import common, config
+from gt4py.next.otf import recipes, stages
+from gt4py.next.program_processors.runners.dace_iterator import (
+    get_sdfg_args,
+    workflow as dace_workflow,
+)
+from gt4py.next.program_processors.runners.gtfn import GTFNBackendFactory
+
+
+class DaCeTranslationStepFactory(factory.Factory):
+    class Meta:
+        model = dace_workflow.DaCeTranslator
+
+
+class DaCeCompilationStepFactory(factory.Factory):
+    class Meta:
+        model = dace_workflow.DaCeCompiler
```

Comment:
Is there a reason to prefer having the factories here, rather than next to their "model"-classes? Think about if you want to add a config option to one of the classes and then update the factory accordingly.